strawberryfields/gbs/subgraph.py

```diff
@@ -205,181 +195,90 @@ def random_search(
-            Value can be either a string selecting from a range of available methods or a
-            customized callable function. Options include:
-
-            - ``"greedy-density"``: resize subgraphs by iteratively adding/removing nodes that
-              result in the densest subgraph among all candidates (default)
-            - ``"greedy-degree"``: resize subgraphs by iteratively adding/removing nodes that
-              have the highest/lowest degree.
-            - *callable*: accepting ``(subgraphs: list, graph: nx.Graph,
-              target: int, resize_options: dict)`` as arguments and returning a ``list``
-              containing the resized subgraphs as lists of nodes, see :func:`greedy_density` for
-              an example signature
-
-    Args:
-        subgraphs (iterable[list[int]]): an iterable over subgraphs, where each subgraph is given
-            by a list of nodes
-        graph (nx.Graph): the input graph
-        target (int): the target number of nodes (must be larger than 1)
-        resize_options (dict[str, Any]): dictionary specifying options used during resizing;
-            defaults to :const:`RESIZE_DEFAULTS`
-
-    Returns:
-        list[list[int]]: a list of resized subgraphs, where each subgraph is given by a list of
-        nodes
-    """
-    resize_options = {**RESIZE_DEFAULTS, **(resize_options or {})}
-
-    try:
-        target = int(target)
-    except TypeError:
-        raise TypeError("target must be an integer")
-
-    if not 2 < target < graph.number_of_nodes():
-        raise ValueError(
-            "target must be greater than two and less than the number of nodes in "
-            "the input graph"
-        )
+def resize(subgraph: list, graph: nx.Graph, sizes: Optional[list] = None) -> list:
+    """Resize a subgraph to a range of input sizes.
 
-    method = resize_options["method"]
+    This function uses a greedy approach to iteratively add or remove nodes one at a time to an
+    input subgraph to reach the range of sizes specified by ``sizes``.
 
-    if not callable(method):
-        method = RESIZE_DICT[method]
-
-    return method(subgraphs=subgraphs, graph=graph, target=target, resize_options=resize_options)
-
-
-def greedy_density(
-    subgraphs: Iterable, graph: nx.Graph, target: int, resize_options: Optional[dict] = None
-) -> list:
-    """Method to greedily resize subgraphs based upon density.
-
-    This function uses a greedy approach to iteratively add/remove nodes to an input subgraph.
-    Suppose the input subgraph has :math:`M` nodes. When shrinking, the algorithm considers
-    all :math:`M-1` node subgraphs of the input subgraph, and picks the one with the greatest
-    density. It then considers all :math:`M-2` node subgraphs of the chosen :math:`M-1` node
-    subgraph, and continues until the desired subgraph size has been reached.
+    When growth is required, the algorithm picks individual nodes from the remainder of the graph
```

Comment:
Will be good to mention how many nodes are examined to be added to the sub-graph in each addition iteration.

Reply:
Great, thanks! Have reworded to hopefully be more clear.